app/controllers/support/providers_controller.rb

```diff
@@ -1,12 +1,28 @@
 module Support
   class ProvidersController < ApplicationController
+    include Pagy::Backend
+
     def index
-      @pagy, @providers = pagy(Provider.order(:provider_name).includes(:courses, :users))
+      @pagy, @providers = pagy(filtered_providers)
     end
 
     def show
       @provider = Provider.find(params[:id])
       render layout: "provider_record"
     end
+
+  private
+
+    def filtered_providers
+      Support::Providers::Filter.call(providers: Provider.order(:provider_name), filters: filters)
```

Comment:
```suggestion
      Support::Providers::Filter.call(providers: RecruitmentCycle.current.poviders.order(:provider_name), filters: filters)
```

We probably only need the current recruitment cycle, else we may need to display it.